### PR TITLE
Make ModelAwareTrait::$modelClass protected.

### PR DESCRIPTION
**Pull request by ADmad**
Allow setting the property to empty string in controller to indicate
that controller does not use a default model.

Alternate to #13178.

### CODE REVIEW

**Comment by ADmad** [comment]
I can add a getter method for the property if deemed necessary.

File: tests/test_app/TestApp/Stub/Stub.php [hunk 1]
```diff
@@ -15,4 +15,9 @@ public function setProps($name)
     {
         $this->_setModelClass($name);
     }
+
+    public function getProp($name)
```
Comment (dereuromark) [inline tests/test_app/TestApp/Stub/Stub.php]: Should we typehint string here too?

Comment (ADmad) [reply tests/test_app/TestApp/Stub/Stub.php]: This is a test case, not worth bothering.

Comment (dereuromark) [reply tests/test_app/TestApp/Stub/Stub.php]: setProps($name) => setProps(?string $name)

Comment (dereuromark) [reply tests/test_app/TestApp/Stub/Stub.php]: My logic is: If it is used wrong, then the expected results are also not what you want.
So probably worth also in those cases. But no biggie.

**Comment by dereuromark** [comment]
I checked. After this one we still have 61 public properties left

**Comment by ADmad** [comment]
The scrutinizer failure is irrelevant, I will fix them on `4.x`.

File: tests/TestCase/Datasource/ModelAwareTraitTest.php [hunk 2]
```diff
@@ -32,10 +32,10 @@ class ModelAwareTraitTest extends TestCase
     public function testSetModelClass()
     {
         $stub = new Stub();
-        $this->assertNull($stub->modelClass);
+        $this->assertNull($stub->getProp('modelClass'));
 
         $stub->setProps('StubArticles');
-        $this->assertEquals('StubArticles', $stub->modelClass);
+        $this->assertEquals('StubArticles', $stub->getProp('modelClass'));
```
Comment (markstory) [inline tests/TestCase/Datasource/ModelAwareTraitTest.php]: You could use `assertAttributeEquals` to avoid having to build your own property reader.

Comment (ADmad) [reply tests/TestCase/Datasource/ModelAwareTraitTest.php]: Done

Comment (dereuromark) [reply tests/TestCase/Datasource/ModelAwareTraitTest.php]: Shouldnt we using Same here? This is safer and will be used more common with scalar comparison.

**Review by dereuromark — APPROVED** [review]

**Pull request merged**